pallets/parachain-staking/src/lib.rs

```diff
@@ -1220,43 +1220,14 @@ pub mod pallet {
 			)
 		}
 
-		/// DEPRECATED use batch util with schedule_revoke_delegation for all delegations
-		/// Request to leave the set of delegators. If successful, the caller is scheduled to be
-		/// allowed to exit via a [DelegationAction::Revoke] towards all existing delegations.
-		/// Success forbids future delegation requests until the request is invoked or cancelled.
-		#[pallet::call_index(19)]
```

Comment:
One thing that is really important is to help make sure we never reuse these call indices. At a minimum, leaving a comment would help, but I think some test cases would be better.

It *might* suffice to leave these indices unused for a full runtime upgrade cycle, but also note that other projects which use `parachain_staking` will have the same potential problem with reuse. The safe choice is clearly to prohibit them from being used again in the future.

Reply:
To elaborate: in the past we had a problem where an extrinsic got recorded in the scheduler with a specific call index, and the pallet shuffled the call indices around during a runtime upgrade. So the extrinsic was decoded and recorded in one runtime and then executed (with its call index misinterpreted) in the next runtime.

These calls probably wouldn't end up in the scheduler, so maybe that's not a problem. It still is safe and harmless to make sure we avoid them in the future, though...

Reply:
Yeah I thought about it and even wanted to write a test, but realized that we need to cherry-pick this PR https://github.com/paritytech/substrate/pull/13558/files into substrate (and maybe maintain until it lands in a release) to get this test to actually get the `call_indices` defined on a pallet. Alternatively we can just always return an error and keep the extrinsic as is.

Reply:
I'd actually be against a no-op since this gives the user a false sense of security when they dispatch it. They might just assume a success for whatever they thought might happen with that dispatchable :) An error is more explicit to call for attention in that regard. 

However, in scope of this PR do we have any opinions if we should/should not add such a function?

Maybe @librelois has an opinion?

Reply:
You think it's also worthwhile to rename the calls to `removed_call_{{index}}`?

Reply:
Thanks updated, please check